Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -1055,6 +1065,7 @@ void MediaSource::onReadyStateChange(ReadyState oldState, ReadyState newState)
         buffer->readyStateChanged();
 
     if (isOpen()) {
+        m_sourceopenPending = false;
```

Comment:
Are there failure cases where `m_sourceopenPending` should be set to `false` to allow the MediaSource wrapper to be collected (outside of when `HTMLMediaElement::detachMediaSource` is called), or will the `sourceopen` event always fire in response to setting `srcObject`?

Reply:
`"sourceopen"` should always fire in response to setting `srcObject`.